crates/motoko/src/lib/vm_types.rs

```diff
@@ -322,25 +342,102 @@ pub struct Agent {
+            cont_source: Source::CoreInit, // todo
+            cont_prim_type,
+        }
+    }
+}
+
+/// A Motoko Actor.
+///
+/// The cost of copying this state is O(1).
+///
+/// An actors has a public API, and can be awaiting many responses as
+/// it services one.  In these ways, an Actor is more complex than the
+/// Motoko Agent that activates it.
+///
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Actor {
     pub store: Store,
-    pub debug_print_out: Vector<crate::value::Text>,
     pub counts: Counts,
+    pub active: Option<Activation>,
+    pub awaiting: HashMap<RespId, Activation>,
+    // to do -- private/public fields/methods.
+}
+
+/// Unique response Id, for coordinating message responses from the
+/// actor replying to the actor or agent receiving the reply.
+/// Generally, an actor is awaiting multiple replies at once, from
+/// other actors.
+#[derive(Clone, Hash, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct RespId(pub usize);
+
+/// A Core encompasses VM system state, including its actors.
+///
+/// The cost of copying this state is O(1).
+///
+/// A VM Core permits an Agent to interact with zero or more Actors.
+///
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Core {
+    pub schedule_choice: ScheduleChoice,
+    pub agent: Agent,
+    pub actors: Actors,
+    pub next_resp_id: usize,
+    pub debug_print_out: Vector<DebugPrintLine>,
+}
+
+/// The current/last/next schedule choice, depending on context.
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub enum ScheduleChoice {
+    Agent,
+    Actor(Id),
+}
+
+/// The Actors in a Core system.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Actors {
+    #[serde(with = "crate::serde_utils::im_rc_hashmap")]
+    pub map: HashMap<Id, Actor>,
+}
+
+/// A line of output emitted by prim "debugPrint".
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct DebugPrintLine {
+    /// Who emitted this output?
+    pub schedule_choice: ScheduleChoice,
+    /// The emitted text.
+    pub text: crate::value::Text,
 }
 
 /// Exclusive write access to the "active" components of the VM.
 pub trait Active: ActiveBorrow {
+    fn schedule_choice<'a>(&'a self) -> &'a ScheduleChoice;
     fn cont<'a>(&'a mut self) -> &'a mut Cont;
     fn cont_source<'a>(&'a mut self) -> &'a mut Source;
     fn cont_prim_type<'a>(&'a mut self) -> &'a mut Option<PrimType>;
     fn env<'a>(&'a mut self) -> &'a mut Env;
     fn stack<'a>(&'a mut self) -> &'a mut Stack;
     fn store<'a>(&'a mut self) -> &'a mut Store;
-    fn debug_print_out<'a>(&'a mut self) -> &'a mut Vector<crate::value::Text>;
+    fn debug_print_out<'a>(&'a mut self) -> &'a mut Vector<DebugPrintLine>;
     fn counts<'a>(&'a mut self) -> &'a mut Counts;
 
     fn alloc(&mut self, value: impl Into<Value_>) -> Pointer {
         self.store().alloc(value)
     }
+
+    fn create(&mut self, _name: Option<Id>, _actor: Actor) -> Result<Value_, Interruption> {
```

Comment:
Create an actor (value) and add it to the VM...

To be useful for Motoko Dev Sever, maybe it should accept a string (read from a file by the caller) and some other stuff, including the canister ID that dfx knows about?

Reply:
BTW -- the name `create` is somewhat arbitrary, but has two nice properties now:
- separate verb from `alloc`, to distinguish it from that case.
- same verb as the IC reference spec, for the act of "creating a canister."